modules/Bio/EnsEMBL/Variation/Pipeline/AncestralAlleles/Init.pm

```diff
@@ -0,0 +1,199 @@
+=head1 LICENSE
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2019] EMBL-European Bioinformatics Institute
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=cut
+
+
+=head1 CONTACT
+ Please email comments or questions to the public Ensembl
+ developers list at <http://lists.ensembl.org/mailman/listinfo/dev>.
+ Questions may also be sent to the Ensembl help desk at
+ <http://www.ensembl.org/Help/Contact>.
+=cut
+
+package Bio::EnsEMBL::Variation::Pipeline::AncestralAlleles::Init;
+
+use strict;
+use warnings;
+
+use FileHandle;
+use Bio::EnsEMBL::Registry;
+use File::Path qw(make_path);
+use Data::Dumper;
+use base qw(Bio::EnsEMBL::Variation::Pipeline::BaseVariationProcess);
+
+sub run {
+  my $self = shift;
+
+  my $run_pipeline = {};
+ 
+  my $compara_dir = $self->param('compara_dir'); 
+  opendir(my $dh, $compara_dir) || die "Can't opendir $compara_dir: $!";
+  my @ancestral_files = grep { $_ ne '.' and $_ ne '..' } readdir($dh);
+  closedir $dh;
+
+  die "There are no ancestral files in $compara_dir" if (scalar @ancestral_files == 0);
+
+  my $registry = 'Bio::EnsEMBL::Registry';
+  my $registry_file = $self->param('ensembl_registry');
+  $registry->load_all($self->param('ensembl_registry'));
+  $self->param('registry', $registry);
+
+  my $vdbas = $registry->get_all_DBAdaptors(-group => 'variation');
+  my $species = {};
+  foreach my $vdba (@$vdbas) {
+      my $species_name = $vdba->species();
+      my $cdba =  $registry->get_DBAdaptor($species_name, 'core');
+      if ($cdba) {
+        $self->param('species', $species_name);
+        my $assembly = $self->get_assembly;
+        my ($ancestral_file) = grep { $_ =~ /$species_name/ && $_ =~ /$assembly/ } @ancestral_files;
+        if ($ancestral_file) {
+          $run_pipeline->{$species_name} = { file => $ancestral_file, assembly => $assembly};
+        }
```

Comment:
An else{} here to report if there is no ancestral file despite there being a core db could be useful.

Reply:
We can also add an entry to meta if the species has ancestral allele data. This should make it more reliable to get all species with ancestral data.